invenio_users_resources/records/api.py

```diff
@@ -211,6 +213,20 @@ def avatar_color(self):
         colors = current_app.config["USERS_RESOURCES_AVATAR_COLORS"]
         return colors[self.id % len(colors)]
 
+    @classmethod
+    def create_via_admin(
+        cls, data, id_=None, validator=None, format_checker=None, **kwargs
+    ):
+        """Create a new User and store it in the database."""
+        try:
+            account_user = current_datastore.create_user(**data)
+            current_datastore.commit()  # Commit to save the user to the database
+            return cls.from_model(account_user)
+        except UniqueConstraintViolation as e:
```

Comment:
Why not capture `IntegrityError` and re-raise it as `ValidationError(message="Already existing username or email", ....)`?
The event listener kind of works, but I think it is a bit overkill. Plus, it will be executed on every commit that happens, which can be quite inefficient.